### DISCUSSION TITLE
A Complete Guideline For Making elementary Icons Dark Variant

### DISCUSSION THREAD

**Discussion by rizmut** [elementary/icons#1088] [Q&A]
### Problem

Fully supported system-wide dark preference [has been being a concern](https://blog.elementary.io/the-need-for-a-freedesktop-dark-style-preference/) for elementaryOS, thus implying a special need for **each element** in the operating system to support the campaign. Unfortunately, for icon themes, I didn't find any kind of directions or guideline to do so.

This requirement is actually intended for LibreOffice applications which require relatively has more icon files compared to ordinary elementaryOS application. In LibreOffice, elementary icon themes in dark UI look suboptimal due to loss of readability and contrast. However, this request for instructions/guideline or perhaps sample files is expected to be applicable to elementaryOS in the long term as well.

Here the screenshot of the issue:

![gambar](https://user-images.githubusercontent.com/5697991/154821529-c373bc7f-ba07-42fe-9a9d-3e78c09a8c37.png)

### Proposal

- Provide a guideline how to make elementary icon in dark variant. This kind of request is not related to symbolic version which has no need for discussion.
- Provide some example files/additional color palettes/etc to do so

**Comment by fitojb** [comment]
The development of a dark variant of the elementary theme for LibreOffice needs an officially-sanctioned, adjusted color palette for dark mode that replaces the darker hues of the light theme with ones that contrast more agains the backdrop. Having this would solve issues in elementary OS’ own UI, like #853.

**Reply by rizmut** [reply]
My current idea is to make the color more gray like the one in the "find and replace" icon in the screenshot of the post. Can this meet the need for sufficient contrast?

It's really regrettable that in LibreOffice, elementary as the default theme for GTK-based desktop doesn't have a proper dark variant while other desktop counterparts already have it all.

**Comment by eyalroz** [comment]
As a staunch supporter of Elementary as a default icon theme for LibreOffice, I - and I am sure, millions if not tens of millions of users - would very much appreciate a Dark mode variant of it. I would encourage both ElementaryOS people and LibreOffice folks to make this happen somehow... and perhaps some funding might also be secured, e.g. from the The Document Foundation (although that is not easy for bureaucratic/regulatory reasons).

**Reply by Wodsfort** [reply]
Agree, this will be good

**Comment by rizmut** [comment]
It turns out that previous default LibreOffice theme: Tango is more clearly visible in dark theme than elementary

<img width="1280" height="602" alt="gambar" src="https://github.com/user-attachments/assets/43cf0faa-54e3-41e0-ae76-99915512ddf7" />

**Comment by newhoa** [comment]
Does LibreOffice support symbolic icons? That would solve a lot of visibility issues since we do provide symbolic icons for Bold, Italic, Underline, Justify, etc. If not, LibreOffice working on symbolic support would help across their entire platform and all themes.

These suggestions are just band-aid suggestions but might help a little for now (sorry, I have no real role in design plans like this).

If LibreOffice doesn't support symbolic icons, maybe just using our symbolic icons but re-colored would be helpful (since svg is text-based, this could be done quickly with a search-replace). This would really help with any of the text or line-based icons I think.

If those screenshots are up to date, some icons could be updated to our current icons (like anything with a [magnifying glass](https://github.com/elementary/icons/blob/main/actions/24/edit-find.svg), or the newer [save icon](https://github.com/elementary/icons/blob/main/actions/16/document-save.svg), both of these are better for dark backgrounds I think).

I never finished these, but if you want you can pull [some of these colorized font-formatting icons](https://github.com/elementary/icons/pull/1289) for now. In addition to that PR, here are a few more I never submitted and never finished working on that you might be able to use:
<img width="16" height="16" alt="format-text-bold (copy 1)" src="https://github.com/user-attachments/assets/88a14d47-e6b1-423b-8ad7-4555637501ec" /> <img width="16" height="16" alt="format-text-italic (copy 2)" src="https://github.com/user-attachments/assets/927c99aa-a11c-42a2-ae66-05dab80b93c9" /> <img width="16" height="16" alt="format-text-underline" src="https://github.com/user-attachments/assets/88bd8b70-7f0e-498a-9229-6ae28e08aa1e" /> <img width="16" height="16" alt="format-text-strikethrough" src="https://github.com/user-attachments/assets/d385a39b-a7f8-4477-8a58-f43e516d1bc4" /> <img width="16" height="16" alt="format-text-none" src="https://github.com/user-attachments/assets/33c567e2-51bb-4a06-8926-388e0038a613" />

I could try to pick back up on some of these if you think they'd help. If so, feel free to add specific icon names and sizes to that PR for ones that are most needed.

Also, check elementary-xfce as I think we've colorized some (like [spell check](https://github.com/shimmerproject/elementary-xfce/blob/master/elementary-xfce/actions/16/tools-check-spelling.svg)). And we still use the old printer icons which I think are more visible on dark themes.

Anyway, not sure if any of this is helpful at all but just thought I'd throw it out there. Till someone better comes along and can do actual design guidelines.
